airflow/utils/sqlalchemy.py

```diff
@@ -92,6 +93,13 @@ def process_result_value(self, value, dialect):
 
         return value
 
+    def load_dialect_impl(self, dialect):
+        if dialect.name == 'mssql':
+            return mssql.DATETIME2(precision=6)
+        elif dialect.name == 'mysql':
+            return mysql.TIMESTAMP(fsp=6)
```

Comment:
I don't think this is the right type for mysql because of the date range.

>  The TIMESTAMP data type is used for values that contain both date and time parts. TIMESTAMP has a range of '1970-01-01 00:00:01' UTC to '2038-01-19 03:14:07' UTC.

Reply:
Currently, in the migration files, we use `TIMESTAMP` because our use of dates is from here: https://github.com/apache/airflow/blob/main/airflow/migrations/db_types.py which applies the `TIMESTAMP` to the databases.  So, we were not really using `DateTime`s in DB but `TIMESTAMP`, unless we want to change it? Maybe in another PR because it's going to be a big change as almost all the date columns uses `TIMESTAMP`